### PR TITLE
Add MyBatis DAO and configure Service layer to use MyBatis

### PR DESCRIPTION
**Pull request by asurelaba**

### CODE REVIEW

File: src/main/resources/mybatis-mapper/customer-mapper.xml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+<mapper namespace="com.solvd.db.dao.interfaces.ICustomerDAO">
+
+    <resultMap id="CustomerResultMap" type="Customer" autoMapping="false">
+        <id property="customerId" column="customer_id"/>
+        <association property="person" column="person_id" javaType="Person"
+                     select="com.solvd.db.dao.interfaces.IPersonDAO.getById"/>
+        <collection property="accounts" column="account_number" ofType="Account" javaType="ArrayList"
+                    resultMap="com.solvd.db.dao.interfaces.IAccountDAO.AccountResultMap" />
+    </resultMap>
```
Comment (asurelaba) [inline src/main/resources/mybatis-mapper/customer-mapper.xml:15]: Is this the right way to mention collections? In this case, i only get the account number in the list but rest of the information about the account is not present. For example
Customer{customerId=4
, person=Person{personId=6, firstName='person6', lastName='person6last', dateOfBirth=1975-11-30, phoneNumber='1231231234', email='person6@abcd.com', address=Address{addressId=6, line1='777 xyz dr', line2='null', zipCode='77777', city=City{cityId=6, cityName='bombay', country=Country{countryId=6, countryName='india'}}}}
, accounts=[Account{
         **accountNumber=20230005, balance=0.0, minBalance=0.0, accountType=null}]**
}

File: src/main/java/com/solvd/db/factory/DAOFactoryManager.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        try {
+            Reader reader = new FileReader(mybatisConfig);
+            Properties properties = new Properties();
+            properties.load(reader);
+            List<String> mybatisList = List.of(properties.getProperty("mybatis").split(","));
+            LOGGER.debug("factory from config file: " + mybatisList + " " + model.getSimpleName());
+            if(mybatisList.contains(model.getSimpleName().toLowerCase())){
+                LOGGER.debug("valid can return mybatis dao factory");
+                return new MyBatisDAOFactory();
+            } else{
```
Comment (ankochergin) [inline src/main/java/com/solvd/db/factory/DAOFactoryManager.java]: reformat code

File: src/main/resources/DAOFactory-config.properties [hunk 3]
```diff
@@ -0,0 +1,2 @@
+factory = mybatis
+mybatis = account,accounttype
```
Comment (ankochergin) [inline src/main/resources/DAOFactory-config.properties]:
```suggestion
factory=mybatis
mybatis=account,accounttype
```

Comment (ankochergin) [reply src/main/resources/DAOFactory-config.properties]: name should be all lowercase

Comment (ankochergin) [inline src/main/resources/DAOFactory-config.properties]: what are this variables for?

Comment (asurelaba) [reply src/main/resources/DAOFactory-config.properties]: These are to configure, which DAOfactory to use either mybatis or jdbc. mybatis property will have list of models that it serves. so we can configure few DAO in jdbc and few from mybatis. Trying to simulate transition from jdbc to mybatis.

Comment (asurelaba) [reply src/main/resources/DAOFactory-config.properties]: they are lowercase. Am i missing something?

Comment (ankochergin) [reply src/main/resources/DAOFactory-config.properties]: sorry, I meant file name

**Pull request merged**